posts/2024-08-28-quarto-on-HPC/index.qmd

```diff
@@ -0,0 +1,222 @@
+
+```
+ln -s /home/$USER/opt/quarto-X.X.X/bin/quarto /home/$USER/bin/quarto
+```
+:::
+
+Check that `quarto` is installed by executing the `quarto --version` command.
+
+# Render and preview documents
+You can render and preview your Quarto markdowns by using the following command from the directory where your `.qmd` files are located.
+
+```
+quarto preview --no-browser --no-watch-inputs
+```
+
+That will render your documents from the markdowns. 
+For `html`, `revealjs`, or `websites` formats, it will start a local server and provide a link that you can paste in your browser to visualize the result.
+
+# Authoring documents with (Python) code cells
+
+To create documents that contain executable Python cells you need `jupyter` installed.
+If you already have `jupyter` in your `base` environment, check if it is being detected by Quarto with the following command.
+```
+quarto check jupyter
+```
+
+That shows the `jupyter` kernel that will be used to execute the Python code cells in your documents.
+
+::: {.callout-caution}
+If you don't already have `jupyter` installed in your `base` environment, Quarto will prompt you to install it.
+
+>| Path: /home/$USER/miniforge-pypy3/bin/python
+>| Jupyter: (None)
+>|
+>| Jupyter is not available in this Python installation.
+>| Install with conda install jupyter
+
+However, you can use `jupyter` from an existing `conda` virtual environment instead of installing it in `base`.
+In case you don't have `jupyter` installed, activate your virtual environment and install `jupyter` as follows (replace `my-venv` with your environment name).
+
+```
+conda activate my-venv
+conda install -c conda-forge jupyter
+```
+
+Then override the `QUARTO_PYTHON` variable to point to the `python` executable in that virtual environment as follows (replace `my-venv` with the name of your virtual environment).
+
+```
+export QUARTO_PYTHON=[...]/envs/my-venv/bin/python
+```
+
+Replace `[...]` in the above command to the absolute path of your `conda` installation.
+In my case it looks like `/home/$USER/miniforge-pypy3`. You can also add that command into your `~/.bashrc` file, so your virtual environment is selected by default when you open a new terminal.
+:::
+
+## Select the `jupyter` kernel
+
+If you are using Quarto with multiple virtual environments, you can select a specific `jupyter` kernel for each document.
+To do that, add `jupyter: my-venv` to the header of your `.qmd` file or replace the exiting `jupyter: pythonX` option if it is already set.
+Then, add the `jupyter` kernel of that virtual environment to the list of available kernels with the following commands.
+
+```
+conda activate my-venv
+ipython kernel install --name "my-venv" --user
+```
+
+::: {.callout-note}
+If you are using `Option 1` for installing Quarto in a `conda` environment, you won't need to specify the kernel since it will use the one available in that virtual environment.
+:::

```

Comment:
Maybe this should be above the section, so those folks know they can skip it?

Reply:
Yeah, I'll move the note above the section, so people who followed Option 1 can skip it.